sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -847,6 +847,7 @@ trait GpuDecimalDivideBase extends GpuExpression {
   def left: Expression
   def right: Expression
   def failOnError: Boolean
+  def failOnDivideByZero: Boolean
```

Comment:
A little confused why this is introduced ? any cases that it is not equal to `failOnError` ?

Reply:
Yes with Average. It is really just a performance improvement. If the count is 0 the output will be null, but I didn't want to have to make divide to a check for 0 in ANSI mode. It is minor.